otter/convergence/service.py

```diff
@@ -283,12 +293,29 @@ def execute_convergence(tenant_id, group_id, build_timeout,
               desired=desired_group_state)
     worst_status, reasons = yield _execute_steps(steps)
 
+    if worst_status != StepResult.LIMITED_RETRY:
+        # If we're not waiting any more, keeping track of the group
```

Comment:
I guess you meant "If we're not waiting any more, **there is no point** keeping track of the group"